### DISCUSSION TITLE
Is there server side streaming?

### DISCUSSION THREAD

**Discussion by 124C41p** [leptos-rs/leptos#716] [Q&A] [answered]
Does Leptos support streaming events from server to client (maybe in form of a server function returning a signal getter (on client side) instead of an ordinary value)? Or is it planned for the future?

This would be really nice for building chat applications or something similar.

**Comment by gbj** [comment, marked as answer]
Yeah take a look at the `<MultiuserCounter/>` component [in this example](https://github.com/leptos-rs/leptos/blob/main/examples/counter_isomorphic/src/counters.rs). It is using server-sent events (SSE) and an EventSource to do the kind of thing you're talking about.

I don't have a plan at the moment to provide additional wrappers for this kind of behavior in the framework but it would be feasible to do in userland or as a library.

**Reply by 124C41p** [reply]
Oh, I must have overlooked that example. Thanks a lot!

**Reply by gbj** [reply]
Yeah `counter_isomorphic` is not necessarily the most obvious name for where you'd find that :-D
